gtsam/base/serialization.h

```diff
@@ -43,122 +43,148 @@
 namespace gtsam {
 
 // Serialization directly to strings in compressed format
-template<class T>
-std::string serialize(const T& input) {
-  std::ostringstream out_archive_stream;
+template <class T>
+void serialize(const T& input, std::ostream& out_archive_stream) {
   boost::archive::text_oarchive out_archive(out_archive_stream);
   out_archive << input;
+}
+
+template <class T>
+void deserialize(std::istream& in_archive_stream, T& output) {
+  boost::archive::text_iarchive in_archive(in_archive_stream);
+  in_archive >> output;
+}
+
+template <class T>
+std::string serialize(const T& input) {
```

Comment:
Added functions for `serializeTo___String` and `serialize___(...) { return serializeTo___String(...); }` and same with deserialize.  Didn't deprecate because I don't see a reason to _not_ use `serialize()`, but lmk if you disagree.

Reply:
Why the underscores? Is that consistent with the Google Style guide?

Reply:
Nvm. I understood what you did here.

Reply:
ya sorry the underscores are just placeholders for nothing ("normal"), XML, or Binary